### PR TITLE
Delete data on logout

### PR DESCRIPTION
**Pull request by ricardobrg**
#258

### CODE REVIEW

File: lib/src/slices/data_bkg/data_bkg_service_email.dart [hunk 1]
```diff
@@ -369,4 +369,14 @@ revolution today.<br />
     return apiOauthInterface != null &&
         await apiOauthInterface.isConnected(account);
   }
+
+  Future<void> deleteAccountData(ApiOAuthModelAccount account) async {
+    List<ApiEmailMsgModel> messages =
```
Comment (mike-audi) [inline lib/src/slices/data_bkg/data_bkg_service_email.dart:374]: other than deleting the corresponding apiAppDataService keys, this logic belongs in the data screen service

File: lib/src/slices/api_google/api_google_service.dart [hunk 2]
```diff
@@ -37,7 +36,6 @@ class ApiGoogleService
     return (await _apiAuthService.getUserInfo(account)) != null;
   }
 
-  @override
```
Comment (mike-audi) [inline lib/src/slices/api_google/api_google_service.dart:40]: this override should not be removed.

File: lib/src/slices/api_app_data/api_app_data_service.dart [hunk 3]
```diff
@@ -28,9 +28,8 @@ class ApiAppDataService {
     if (data != null) await _repository.delete(data.id!);
   }
 
-  Future<void> logout() async {
+  Future<void> deleteEmailData() async {
```
Comment (mike-audi) [inline lib/src/slices/api_app_data/api_app_data_service.dart:31]: this logic doesn't belong in this service, this service should just add/update/delete whatever its told. it shouldn't know which keys belong to what. the logic to delete the background fetch keys (label, epoch, page) belong in the data background service.  in the apiAppData service there should be a function called deleteAll which deletes ALL keys on logout

File: lib/src/slices/data_bkg/data_bkg_service_email.dart (hunk 1 above)
Comment (ricardobrg) [reply lib/src/slices/data_bkg/data_bkg_service_email.dart:374]: the data screen service doesn't use the ApiEmailMsgService and ApiEmailSenderService... that's why I have implemented it here...

**Pull request merged**